apps/molgenis-components/src/components/forms/ButtonDropdown.vue

```diff
@@ -1,65 +1,123 @@
 <template>
-  <span>
-    <div class="dropdown m-0 p-0" :class="{ show: display }">
-      <button
-        class="btn btn-outline-primary border-0"
-        :class="{
-          'dropdown-toggle': !icon,
-          'nav-link': isMenuItem,
-        }"
-        type="button"
-        data-toggle="dropdown"
-        aria-haspopup="true"
-        aria-expanded="false"
-        @click="toggle"
-      >
-        <span v-if="label">{{ label }}</span>
-        <span v-if="icon" :class="'fa fa-' + icon + ' fa-lg ml-2'"></span>
-      </button>
-    </div>
-    <div
+  <span class="dropdown">
+    <button
+      type="button"
+      ref="showInputButton"
+      class="btn btn-outline-primary"
+      :class="{ 'dropdown-toggle': !icon }"
+      @click="toggle"
+    >
+      <span v-if="label">{{ label }}</span>
+      <span v-if="icon" :class="'fa fa-' + icon + ' fa-lg ml-2'"></span>
+    </button>
+    <span
       v-if="display"
-      class="dropdown-menu bg-white"
-      style="position: absolute; z-index: 100"
-      :class="{ show: display }"
+      class="bg-white border rounded"
+      ref="dropdown"
       v-click-outside="toggle"
     >
-      <div class="form-group dropdown-item">
-        <slot :close="toggle" />
-      </div>
-    </div>
+      <slot :close="toggle" />
+    </span>
   </span>
 </template>
 
 <script>
 import vClickOutside from "v-click-outside";
+import Popper from "popper.js";
 
 export default {
   directives: {
     clickOutside: vClickOutside.directive,
   },
   props: {
     label: String,
-    icon: String,
+    icon: {
+      type: String,
+      required: false,
+      default: "",
+    },
     isMenuItem: Boolean,
+    placement: {
+      type: String,
+      required: false,
+      default: () => "bottom-start",
+    },
   },
   data() {
     return {
       display: false,
+      popperInstance: null,
     };
   },
   methods: {
-    toggle() {
+    async toggle() {
       this.display = !this.display;
+      if (this.display) {
+        await this.$nextTick();
+        const dropDownBtn = this.$refs["showInputButton"];
```

Comment:
looks rather error prone to use refs. Now there is a lot a dev needs to know while making changes? Is there not a better way?